apps/aecore/lib/aecore/oracle/oracle_state_tree.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Aecore.Oracle.OracleStateTree do
+  @moduledoc """
+  Top level oracle state tree.
+  """
+  alias Aeutil.PatriciaMerkleTree
+  alias Aeutil.Serialization
+  alias Aecore.Oracle.Tx.OracleQueryTx, as: Query
+
+  @type oracles_state :: %{oracle_tree: Trie.t(), oracle_cache_tree: Trie.t()}
+  @dummy_val <<0>>
+
+  @spec init_empty() :: oracles_state()
+  def init_empty do
+    %{
+      oracle_tree: PatriciaMerkleTree.new(:oracles),
+      oracle_cache_tree: PatriciaMerkleTree.new(:oracles_cache)
+    }
+  end
+
+  @spec prune(Chainstate.t(), non_neg_integer()) :: Chainstate.t()
+  def prune(chainstate, block_height) do
+    {new_oracles_state, new_accounts_state} =
+      initialize_deletion({chainstate.oracles, chainstate.accounts}, block_height - 1)
```

Comment:
this seems like oracle logic, not tree logic as well

Reply:
Refer to: https://github.com/aeternity/elixir-node/pull/426#issuecomment-401778877